pkg/identity/keystone/token_getter.go

```diff
@@ -89,10 +92,20 @@ func GetToken(options Options) (*tokens3.Token, error) {
 
 	if setTransport {
 		transport := &http.Transport{Proxy: http.ProxyFromEnvironment, TLSClientConfig: tlsConfig}
-		client.HTTPClient.Transport = transport
+		provider.HTTPClient.Transport = transport
 	}
 
-	v3Client, err := openstack.NewIdentityV3(client, gophercloud.EndpointOpts{})
+	if klog.V(6) {
+		if provider.HTTPClient.Transport == nil {
+			provider.HTTPClient.Transport = http.DefaultTransport
+		}
+		provider.HTTPClient.Transport = &client.RoundTripper{
```

Comment:
Don't need `else` here?

Reply:
no, since we embed `provider.HTTPClient.Transport` into a RoundTripper.